airflow-core/src/airflow/models/dag.py

```diff
@@ -1917,7 +1918,7 @@ class DagModel(Base):
     # Asset expression based on asset triggers
     asset_expression = Column(sqlalchemy_jsonfield.JSONField(json=json), nullable=True)
     # DAG deadline information
-    deadline = Column(sqlalchemy_jsonfield.JSONField(json=json), nullable=True)
+    _deadline = Column("deadline", sqlalchemy_jsonfield.JSONField(json=json), nullable=True)
```

Comment:
This change allows us to change the python name of the column without changing the actual SQL name, so no migration is needed.  I did it this way so the property (below) can use the existing name and allow for proper serialization/deserialization.